### PR TITLE
serve-http2-test-server: Support h2c connections

### PR DESCRIPTION
**Pull request by Miciah**
Allow clients to connect to the HTTP/2 test server using cleartext HTTP/2 (h2c).

* `test/http2/server.go`: Support h2c connections.
* `go.mod`:
* `go.sum`:
* `vendor/*`:
* `vendor/modules.txt`: Regenerate.

### CODE REVIEW

File: test/http2/server.go [hunk 1]
```diff
@@ -37,7 +40,13 @@ func Serve() {
 		port := lookupEnv("HTTP_PORT", defaultHTTPPort)
 		log.Printf("Listening on port %v\n", port)
 
-		if err := http.ListenAndServe(":"+port, nil); err != nil {
+		h2s := &http2.Server{}
+		h1s := &http.Server{
```
Comment (frobware) [inline test/http2/server.go:44]: nit: is this http1? Or still http2 but with cleartext?

**Comment by frobware** [comment]
/hold
@Miciah I added a hold as a gofmt needs to run. This is just spinning in CI.

File: test/http2/server.go (hunk 1 above)
Comment (Miciah) [reply test/http2/server.go:44]: It's cleartext HTTP/1.  I'll add comments to clarify.

**Comment by Miciah** [comment]
> @Miciah I added a hold as a gofmt needs to run. This is just spinning in CI.

Thanks!  Fixed.

**Comment by Miciah** [comment]
/close
Not needed for https://github.com/openshift/origin/pull/26456.  We can resurrect this PR if we ever need it for some other test.

**Pull request closed without merging**